client/my-sites/plans/current-plan/jetpack-checklist/index.tsx

```diff
@@ -158,17 +165,20 @@ class JetpackChecklist extends PureComponent< Props > {
 				<Checklist
 					className="jetpack-checklist"
 					isPlaceholder={ ! taskStatuses }
+					onExpandTask={ this.trackExpandTask }
 					progressText={ translate( 'Your Jetpack setup progress' ) }
 				>
 					<Task
 						{ ...JETPACK_CHECKLIST_TASK_PROTECT }
+						id="jetpack_task_protect"
```

Comment:
I tried moving these `id`s to `./constants.ts`, but that's anything but trivial, due to the vastly heterogenous handling of task IDs: e.g. for some IDs, we use object keys in a collection of tasks -- but not all tasks are contained in a collection. Since the constants file is typed, we'd have to add an `id: string?` field, i.e. a conditional ID, which would be _very_ atypical for an ID. (In other words, typing the constants file reveals that there's too much heterogenity in how we handle our tasks.)

My conclusion is thus that we should probably either move all task definitions from this file to `./constants.ts`, or vice versa (merge `./constants.ts` into this file). Either way, that's for another PR.

Reply:
I don't understand why the tasks live spread across two files either. I do think it's OK to add the option `id`. Types should be serving us by providing better insights and safety. We don't want them to strictly get in the way.

This may be types highlighting a situation that is hard to justify. That's relevant and something we should step back and handle as you suggest. That said, I don't think the fact that types exist should prevent us from describing an existing sub-optimal situation. They can continue to help us describe the situation as it is now. In the future when we come to an improved longer-term solution they'll help us to clean up as well 🙂 

```patch
diff --git a/client/my-sites/plans/current-plan/jetpack-checklist/constants.ts b/client/my-sites/plans/current-plan/jetpack-checklist/constants.ts
index 1c19143853..954d7a6ed0 100644
--- a/client/my-sites/plans/current-plan/jetpack-checklist/constants.ts
+++ b/client/my-sites/plans/current-plan/jetpack-checklist/constants.ts
@@ -19,6 +19,7 @@ export function getJetpackChecklistTaskDuration( minutes: number ): string {
 }
 
 interface TaskUiDescription {
+	readonly id?: string;
 	readonly title: string;
 	readonly description?: string;
 	readonly completedButtonText: string;
@@ -105,6 +106,7 @@ export const JETPACK_CHECKLIST_TASK_AKISMET: TaskUiDescription = {
 };
 
 export const JETPACK_CHECKLIST_TASK_PROTECT: TaskUiDescription = {
+	id: 'jetpack_task_protect',
 	title: translate( "We're automatically protecting you from brute force login attacks." ),
 	completedTitle: translate( "We've automatically protected you from brute force login attacks." ),
 	completedButtonText: translate( 'Configure' ),
diff --git a/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx b/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx
index cb0d1dfdf1..57d261db29 100644
--- a/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx
+++ b/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx
@@ -170,7 +170,6 @@ class JetpackChecklist extends PureComponent< Props > {
 				>
 					<Task
 						{ ...JETPACK_CHECKLIST_TASK_PROTECT }
-						id="jetpack_task_protect"
 						completed
 						href={ JETPACK_CHECKLIST_TASK_PROTECT.getUrl( siteSlug ) }
 						onClick={ this.handleTaskStart( { taskId: 'jetpack_protect' } ) }
```